controller/users.go

```diff
@@ -580,6 +580,33 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 			}
 		}
 
+		updatedFeatureLevel := ctx.Payload.Data.Attributes.FeatureLevel
+		if log.IsDebug() {
+			currentFeatureLevel := "none"
+			newFeatureLevel := "none"
+			if user.FeatureLevel != nil {
+				currentFeatureLevel = *user.FeatureLevel
+			}
+			if updatedFeatureLevel != nil {
+				newFeatureLevel = *updatedFeatureLevel
+			}
+			log.Debug(ctx, map[string]interface{}{"current_feature_level": currentFeatureLevel, "new_feature_level": newFeatureLevel}, "updating feature level")
+		}
+		if updatedFeatureLevel != nil && (user.FeatureLevel == nil || *updatedFeatureLevel != *user.FeatureLevel) {
+			// handle the case where the value needs to be reset, when the new value is "" (empty string) or "released"
+			if *updatedFeatureLevel == "" || *updatedFeatureLevel == "released" {
+				log.Debug(ctx, map[string]interface{}{"user_id": user.ID}, "resetting feature level")
+				user.FeatureLevel = nil
+			} else {
+				// if the level is 'internal', we need to check against the email address to verify that the user is a Red Hat employee
+				if *updatedFeatureLevel == "internal" && !strings.HasSuffix(user.Email, "@redhat.com") {
```

Comment:
1. We also need to check if email is verified. See https://github.com/fabric8-services/fabric8-auth/pull/227/files#diff-c5cd9222ed04372b0880450d7dc3e71bR35 (that PR is not merged yet).

2. I really don't like @redhat.com to be hardcoded here... Let's create a configuration for internal email suffix and set it to "@redhat.com" by default so we don't have to configure it for OSIO. But at least it will be possible to use something else in a custom configuration.

Reply:
yes, I intend to use the `EmailVerified` field once it's in place. Let me put a `TODO` here for now. And yes, let me use a config variable to externalize the email address domain, that will be cleaner, indeed.